addons/l10n_ch/models/res_bank.py

```diff
@@ -272,7 +272,8 @@ def _is_qr_iban(self):
         QR-codes. They are formed like regular IBANs, but are actually something
         different.
         """
-        self.ensure_one()
+        if len(self) != 1:
+            return UserError(_("Bank account is mandatory to generate QR invoice"))
```

Comment:
I understand why you do that but your error message doesn't correspond totally to the if statement. What happened if you pass a recordset of 2 `res.partner.bank`? It seems that it couldn't happen in the flow but please adapt your error message.

Simple suggestion:
```suggestion
            return UserError(_("You must have one and only one bank account to generate QR invoice."))
```

Btw, perhaps you have to check before that you only have one bank account?